components/mediation-admin/org.wso2.carbon.mediation.templates/src/main/java/org/wso2/carbon/mediation/templates/services/EndpointTemplateEditorAdmin.java

```diff
@@ -262,7 +287,14 @@ private void saveEndpointTemplate(OMElement templateElement) throws AxisFault {
                     Template templ = config.getEndpointTemplates().get(templateName);
                     if (templ != null) {
 //                        templ.init(getSynapseEnvironment());
-                        persistTemplate(templ);
+                        CAppArtifactDataService cAppArtifactDataService = ConfigHolder.getInstance().
+                                getcAppArtifactDataService();
+                        if (cAppArtifactDataService.isArtifactDeployedFromCApp(getTenantId(), ServiceBusConstants.TEMPLATE_TYPE + File.separator + templateName)) {
+                            cAppArtifactDataService.setEdited(getTenantId(), ServiceBusConstants.TEMPLATE_TYPE + File.separator + templateName);
```

Comment:
Use variable for 'ServiceBusConstants.TEMPLATE_TYPE + File.separator + templateName' and use it in multiple places